src/com/palmergames/bukkit/towny/event/town/TownPreUnclaimEvent.java

```diff
@@ -0,0 +1,98 @@
+package com.palmergames.bukkit.towny.event.town;
+
+import com.palmergames.bukkit.towny.object.Town;
+import com.palmergames.bukkit.towny.object.TownBlock;
+import com.palmergames.bukkit.towny.object.Translation;
+import org.bukkit.Bukkit;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.NotNull;
+
+
+public class TownPreUnclaimEvent extends Event implements Cancellable {
+
+    private static final HandlerList handlers = new HandlerList();
+    private final TownBlock townBlock;
+    private final Town town;
+    private boolean isCancelled = false;
+    private String cancelMessage = Translation.of("msg_err_town_unclaim_canceled");
+
+    @NotNull
+    @Override
+    public HandlerList getHandlers() {
+        return getHandlerList();
+    }
+    
+    public static HandlerList getHandlerList() {
+		return handlers;
+	}
+
+    /**
+     * Event thrown prior to a {@link TownBlock} being unclaimed by a {@link Town}.
+     * <p>
+     * This is cancellable but it is probably not a good idea to do
+     * so without testing.
+     *
+     * @param town The Town unclaiming the TownBlock.
+     * @param townBlock The TownBlock that will be unclaimed.
+     */
+    public TownPreUnclaimEvent(Town town, TownBlock townBlock) {
+        super(!Bukkit.getServer().isPrimaryThread());
+        this.town = town;
+        this.townBlock = townBlock;
+
+        // Don't even bother with things if town is null.
+        if (this.town == null){
+            this.cancelMessage = Translation.of("msg_area_not_recog");
+            setCancelled(true);
+        }
+    }
+
+    /**
+     * @return Check if the event is cancelled or not, yet.
+     */
+    @Override
+    public boolean isCancelled() {
+        return isCancelled;
+    }
+
+    /**
+     * @param cancelled The incoming boolean that sets if the event is to be cancelled.
+     */
+    @Override
+    public void setCancelled(boolean cancelled) {
+        isCancelled = cancelled;
+    }
+    
+    /**
+     * @return the {@link Town}.
+     */
+    public Town getTown() {

```

Comment:
This should get an @nullable annotation on it to handle the cases where townless-townblocks are being removed, or alternatively, no event should be fired if the removeTownBlock() method has a null town.

Reply:
Given that there's 28 calls to removeTownBlock() (not counting instances in comments and the changelog), I've opted for the shorter route.